docker/datahub-ingestion-base/Dockerfile

```diff
@@ -5,41 +5,52 @@ ARG BASE_IMAGE=base
 ARG ALPINE_REPO_URL=http://dl-cdn.alpinelinux.org/alpine
 ARG GITHUB_REPO_URL=https://github.com
 ARG DEBIAN_REPO_URL=https://deb.debian.org/debian
+ARG UBUNTU_REPO_URL=http://ports.ubuntu.com/ubuntu-ports
 ARG PIP_MIRROR_URL=https://pypi.python.org/simple
 
-FROM powerman/dockerize:0.19 as dockerize-binary
+FROM powerman/dockerize:0.19 AS dockerize-binary
 
-FROM python:3.10 AS base
+FROM ubuntu:22.04 AS base
 
 ARG GITHUB_REPO_URL
 
 ENV DEBIAN_FRONTEND=noninteractive
 
 # Optionally set corporate mirror for deb
 ARG DEBIAN_REPO_URL
-RUN if [ "${DEBIAN_REPO_URL}" != "http://deb.debian.org/debian" ] ; then sed -i "s#http.*://deb.debian.org/debian#${DEBIAN_REPO_URL}#g" /etc/apt/sources.list.d/debian.sources ; fi
+ARG UBUNTU_REPO_URL
+RUN if [ "${DEBIAN_REPO_URL}" != "http://deb.debian.org/debian" ] ; then sed -i "s#http.*://deb.debian.org/debian#${DEBIAN_REPO_URL}#g" /etc/apt/sources.list ; fi
+RUN if [ "${UBUNTU_REPO_URL}" != "http://ports.ubuntu.com/ubuntu-ports" ] ; then sed -i "s#http.*://ports.ubuntu.com/ubuntu-ports#${UBUNTU_REPO_URL}#g" /etc/apt/sources.list ; fi
 
 # Optionally set corporate mirror for pip
 ARG PIP_MIRROR_URL
 RUN if [ "${PIP_MIRROR_URL}" != "https://pypi.python.org/simple" ] ; then pip config set global.index-url ${PIP_MIRROR_URL} ; fi
 ENV UV_INDEX_URL=${PIP_MIRROR_URL}
 
-RUN apt-get update && apt-get install -y -qq \
+RUN apt-get update && apt-get upgrade -y \
+    && apt-get install -y -qq \
+    python3 \
```

Comment:
can we make it explicitly depend on python 3.10?

Reply:
btw can we try adding `--no-install-recommends` to `apt-get install`? i suspect it could help reduce image size and number of vulnerable packages